### ISSUE TITLE
[Feature] PostgreSQL Support

### ISSUE THREAD

**Issue by benthecarman** [cloudflare/workers-rs#386]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Description

Recently you guys have announced support for various database integrations

https://blog.cloudflare.com/announcing-database-integrations/

https://developers.cloudflare.com/workers/tutorials/postgres

However it is only available for workers in javascript which is untenable for a lot of users.

Would be great if support was added to the rust side

**Comment by janpio** [comment]
Isn't that something that would be needed to be fixed in a PostgreSQL driver like `tokio-postgres`, similar to how it was changed with `pg` via contribution? Or is the underlying API not surfaced in Rust (yet)?

**Comment by kflansburg** [comment]
I believe that we support tokio-postgres after this PR: https://github.com/sfackler/rust-postgres/pull/1040

The current issue is that only plaintext is supported, TLS support is more complex due to the use of StartTLS. We support StartTLS, but tokio postgres would need modification to use this API.

**Comment by benthecarman** [comment]
> I believe that we support tokio-postgres after this PR: [sfackler/rust-postgres#1040](https://github.com/sfackler/rust-postgres/pull/1040)
> 
> The current issue is that only plaintext is supported, TLS support is more complex due to the use of StartTLS. We support StartTLS, but tokio postgres would need modification to use this API.

Yes I found that, however, because of the no tls it is pretty much unusable for anything in the real world

**Comment by aviplayer** [comment]
Any way to provide custom certificate?

**Comment by kflansburg** [comment]
> Is there any way to provide custom certificate?

TCP sockets for Workers does not currently support custom certificates, but this is on our roadmap.
